src/prototype_room_creepbuilder.js

```diff
@@ -117,23 +117,49 @@ Room.prototype.spawnCheckForCreate = function(creepsConfig) {
 
     let creep = this.memory.queue[0];
     energyNeeded = 50;
-
-    if (this.spawnCreateCreep(creep.role, creep.heal, creep.level, creep.squad, creep.routing, creep.base)) {
+    let c_room = this;    
+    let sortByDistance = function(object) {
+      return Game.map.getRoomLinearDistance(c_room.name, object);
+    };
+    let roomsMy = _.sortBy(Memory.myRooms, sortByDistance);
+    if (this.spawnCreateCreep(creep.role, creep.heal, creep.level, creep.squad, creep.routing, this.name)) {
+      console.log('['+this.name +' Spawning]- Activating '+creep.role );
       this.memory.queue.shift();
-    } else {
-      if (creep.ttl === 0) {
+    } else if (creep.ttl) {
+      if ( creep.ttl === 0) {
+        for (var room_id in roomsMy) {
+          var s_room = Game.rooms[roomsMy[room_id]];
+          if (s_room !== this) {
```

Comment:
You should filter the rooms, also for rooms too far away. Many of my rooms can not reach some other rooms. In this step also remove `this` room. And I would do that before ordering.